client/my-sites/themes/controller.js

```diff
@@ -125,10 +126,11 @@ export function details( context, next ) {
 		isLoggedIn: !! user
 	};
 
-	context.store.dispatch( setSection( 'themes', {
-		hasSidebar: false,
+	// When we're logged in, we need to remove the sidebar.
+	removeSidebar( context, {
+		section: 'themes',
 		isFullScreen: true
-	} ) );
+	} );
```

Comment:
Careful, we're only `dispatch`ing `setSection` here. For the actual unmounting, we take this declarative [detour](https://github.com/Automattic/wp-calypso/pull/3513/files#diff-85f90b61d9c36634c5464a3180470726R136) in order to be able to unmount only when rendering to the client, but not on the server. So this call should remain as it is :-)